### PR TITLE
Update Hera site config, update documentation for Hera and (AWS) Ubuntu 20

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR:
- updates the Hera site config for `skylab-2.0.0` and adds information on how to create and use a `spack` mirror on this system
- updates the temporary AWS Parallelcluster instructions in `configs/sites/aws-pcluster/README.md` with the latest changes
- updates the documentation for all platforms for `skylab-2.0.0`

No code changes, only documentation. The updated documentation can be accessed from the system checks (https://spack-stack--367.org.readthedocs.build/en/367/Platforms.html#pre-configured-sites).

### CODE REVIEW

File: doc/source/MaintainersSection.rst [hunk 1]
```diff
@@ -172,16 +172,41 @@ ecflow
 NOAA RDHPCS Hera
 ------------------------------
 
+On Hera, ``miniconda`` must be installedas a one-off before spack can be used.
```
Comment (srherbener) [inline doc/source/MaintainersSection.rst]:
```suggestion
On Hera, ``miniconda`` must be installed as a one-off before spack can be used.
```

File: doc/source/Platforms.rst [hunk 2]
```diff
@@ -282,26 +308,6 @@ The following is required for building new spack environments and for using spac
    module load miniconda/3.9.12
    module load ecflow/5.8.4
 
-For ``spack-stack-1.0.1`` with Intel, load the following modules after loading miniconda and ecflow:
-
-   ulimit -s unlimited
```
Comment (srherbener) [inline doc/source/Platforms.rst:287]: Why do we list `ulimit -s unlimited` in some recipes and not in others. Don't we always want to use "unlimited" stack size?

**Review by srherbener — COMMENTED** [review]
The new formatted page looks great! Thanks! I just have a couple minor comments.

File: doc/source/Platforms.rst (hunk 2 above)
Comment (climbfuji) [reply doc/source/Platforms.rst:287]: You are right, I removed it everywhere except where it's listed as recommendation for generic Linux/macOS.

Comment (srherbener) [reply doc/source/Platforms.rst:287]: Sounds great. thanks!

**Review by srherbener — APPROVED** [review]
Thanks for addressing my comments!

**Review by AlexanderRichert-NOAA — COMMENTED** [review]
What's the story behind blacklisting the ecflow module?

**Comment by climbfuji** [comment]
> What's the story behind blacklisting the ecflow module?

Good question! There is the `ecflow` module that the sysadmins created and that needs to be loaded. If spack creates its own module for an external package, it is basically empty and `ecflow` won't run.

**Review by AlexanderRichert-NOAA — APPROVED** [review]
Haven't tested hera changes, but everything looks reasonable

**Comment by climbfuji** [comment]
Since this PR doesn't touch anything except documentation and site configs for hera/aws, we don't need to rerun the failed CI tests (they failed because the ftp server hosting wgrib2 was down/inaccessible).

**Pull request merged**